test/main.cc

```diff
@@ -0,0 +1,151 @@
+#include <random>
+
+#include "../src/class.h"
+#include "gtest/gtest.h"
+
+const int kDefaultWidth = 80;
+const int kDefaultHeight = 60;
+
+class WebPSamplesGenerator {
+ public:
+  // Initializing.
+  WebPSamplesGenerator()
+      : pic_count(10),
+        width(kDefaultWidth),
+        height(kDefaultHeight),
+        transparency(0xff),
+        randomized(true) {}
+
+  WebPSamplesGenerator(int pic_count, uint8_t transparency, bool randomized)
+      : pic_count(pic_count),
+        width(kDefaultWidth),
+        height(kDefaultHeight),
+        transparency(transparency),
+        randomized(randomized) {}
+
+  WebPSamplesGenerator(int pic_count, int width, int height,
+                       uint8_t transparency, bool randomized)
+      : pic_count(pic_count),
+        width(width),
+        height(height),
+        transparency(transparency),
+        randomized(randomized) {}
+
+  // Returns RGBA values for WebPPicture.
+  std::vector<uint8_t> GenerateRGBA(int seed) {
+    std::mt19937 rng(seed);
+    std::vector<uint8_t> rgba;
+
+    int color_R = rng();
+    int color_G = rng();
+    int color_B = rng();
+
+    for (int i = 0; i <= height; ++i) {
+      for (int j = 0; j <= width * 4; ++j) {
+        rgba.push_back(randomized ? rng() & 0xff : color_R);
+        rgba.push_back(randomized ? rng() & 0xff : color_G);
+        rgba.push_back(randomized ? rng() & 0xff : color_B);
+        rgba.push_back(transparency);
+      }
+    }
+    return rgba;
+  }
+
+  // Returns vector of WebPPicture(s).
+  // randomized == true: picture is random noise
+  // randomized == false: picture is random solid color
+  std::vector<std::unique_ptr<WebPPicture, void (*)(WebPPicture*)>>
+  GeneratePics() {
+    std::vector<std::unique_ptr<WebPPicture, void (*)(WebPPicture*)>> pics;
+    for (int i = 0; i < pic_count; ++i) {
+      pics.emplace_back(new WebPPicture, WebPPictureFree);
```

Comment:
This creates a memory leak because WebPPictureFree frees the WebPPicture's internal data, but it doesn't actually free the WebPPicture pointer. If you want to keep this approach, you could create a small function that calls WebPPictureFree then deletes the pointer.